rfcs/text/0000-contributed-pipelines.md

```diff
@@ -0,0 +1,161 @@
+
+#### Why Terra?
+Contributing pipelines via Terra has a few benefits:
+1. Ensures the pipeline can be run in the DCP pipeline infrastructure
+2. Provides support for contributors developing and testing the pipelines
+3. Makes the pipelines immediately available to the community, where users can run the pipelines on HCA data, in the case where they may be useful but not immediately able to be accepted to the DCP, for example if they haven’t met an acceptance criteria, or aren’t prioritized into production because of a reason in section 3.3
+4. Enables the contributor to test the pipeline on HCA data and demonstrate functionality and performance
+
+The guidelines drafted above describe the contribution process as user facing. When we publish them officially, we will include links to tutorials and more background on the contribution requirements outlined in this document.
+
+### Prioritizing contributions
+It will take some effort on behalf of DCP's engineers and computational biologists to assist external developers in adapting their pipelines for use in the DCP. As a result, the order of incorporation for contributed pipelines will consider factors like:
+
+- The amount of data in the DCP for that assay.
+- The rate at which new data for that assay is being added, based on the HCA Data Roadmap.
+- Relative difficulty of adapting the pipeline to run in the DCP.
+- Value to the community in exposing this data to users (can it be made available in a different service?)
+- Risk associated with inaction (will we lose the ability to include data in the reference atlas?)
+- The amount of external developer bandwidth to support pipeline incorporation.
+- HCA Community feedback (for example, polls of the HCA community that ask them to rank the 5 assays they think will be most important in the coming year).
+- Whether there is an existing pipeline in the HCA that serves the same data as the contributed pipeline.
+- Support for this data type in HCA metadata.
+- Inclusion of the pipeline in the HCA Scientific Roadmap or communication from HCA scientific stakeholders that the assay is a priority.
+
+When a pipeline is prioritized for incorporation into the DCP, the contributor will be contacted and reminded of the requirements outlined here prior to running in production. The pipeline will be documented on the HCA Data Portal, where the author will be cited for their contribution. Additionally, analysis data produced by the pipeline will be labeled with minimally the pipeline author and contact info, and a permanent link to the pipeline reference workspace in the HCA analysis metadata.
+
+## Operating pipelines
+### Responding to Failures
+Operating any DCP pipeline on relevant data may result in occasional pipeline failures. The pipelines team will use the following procedure to resolve these failures:
```

Comment:
I'm a little nervous that the wording around this section implies that pipeline failures should be dealt with "as soon as possible." The reality is that as you grow the number of pipelines, you are going to get more failures and each failure is going to take longer to debug because there are increasing number of code paths that could have caused the issue. 

We need some notion of SLOs that sets expectations on when a user might be able to expect data processed by a broken pipeline (that is hopefully pretty generous) and prioritize AWG-vetted pipeline fixes above anything else. I'm not sure if there's a particular team that will be delegated to pipeline maintenance but the eng team will likely have to juggle maintenance with other priorities in the DCP which is something to be mindful of.